ViTool/ViewModel/TranslateXmlToTxTViewModel.cs

```diff
@@ -0,0 +1,248 @@
+
+        private string _Output;
+        public string Output
+        {
+            get { return _Output; }
+            set
+            {
+                if (_Output == value)
+                    return;
+
+                if (value.Length > maxOutputLines)
+                    _Output = "";
+                else
+                    _Output = value;
+            }
+        }
+        /// <summary>
+        /// 0 - 100
+        /// </summary>
+        public int ProgressPercent { get; set; }
+        public int EstimatedTime { get; set; }
+        public string SelectedClass { get; set; }
+        public string NewClass { get; set; }
+        public ObservableCollection<string> ListOfClasses { get; set; }
+
+        #endregion
+
+        #region CTOR
+        public TranslateXmlToTxTViewModel(TranslateXmlToTxTAlgorithm translateXmlToTxT, Progress<ProgressReportModel> progress, IndicatorColors indicatorColors, Settings settings)
+        {
+            this.settings = settings;
+            this.progress = progress;
+            this.TranslateXmlToTxT = translateXmlToTxT;
+            this.indicatorColors = indicatorColors;
+
+            SettingsData currentSettingsData = settings.ReadSettings();
+
+            EstimatedTime = 0;
+            ProgressPercent = 0;
+            TranslateXmlToTxTSrc = currentSettingsData.LastOpenedDirectory;
+            TranslateXmlToTxTInfoBrush = new SolidColorBrush(Color.FromRgb(220, 220, 220));
+            ListOfClasses = new ObservableCollection<string>();
+
+            foreach (string newClass in currentSettingsData.SavedClasses)
+                ListOfClasses.Add(newClass);
+
+                //new ObservableCollection<string>() { "HCH", "LowFreqAnomaly", "Imprint", "Break", "ChippedBreak" };
+
+            progress.ProgressChanged += ReportProgress;
+        }
+        #endregion
+
+        #region Commands
+        private RelayCommand _CreateTxtFromXml;
+        public RelayCommand CreateTxtFromXml
+        {
+            get
+            {
+                if (_CreateTxtFromXml == null)
+                {
+                    _CreateTxtFromXml = new RelayCommand(
+                    async () =>
+                    {
+                        Output = "";
+
+                        SettingsData currentSettingsData = settings.ReadSettings();
+                        TranslateXmlToTxTInfoBrush = indicatorColors.busyColor;
+                        TranslateXmlToTxTSrc = selectPath(currentSettingsData.LastOpenedDirectory, "Point to folder with xml files. \nProgram will create txt files called 'yourFile.txt' next original ones.");
+
+                        if (TranslateXmlToTxTSrc == null || TranslateXmlToTxTSrc == "")
+                        {
+                            TranslateXmlToTxTSrc = "No Directory";
+                            TranslateXmlToTxTInfoBrush = indicatorColors.errorColor;
+                            Output += "There is no files";
```

Comment:
i think it would be better to use string builder here, remember if u add string to string you allocate new memory to heap